### PR TITLE
Feature flags using react context api

### PR DESCRIPTION
**Pull request by seanfeld**
# React Context Feature Flags

## Description
Adds feature flags using the react context api and adds the feature flag support for replication transformations. This also shows how to use the feature flag hook from inside a class component using the higher order component pattern.

Unfortunately CDAP UI currently gets all configuration from the window in `server/express.js` so making use of that to have these feature flags from the backend.

I also removed `<If></If>` from main.js while working on other things. 

## PR Type
- [ ] Bug Fix
- [x] Feature
- [ ] Build Fix
- [ ] Testing
- [ ] General Improvement
- [ ] Cherry Pick

### CODE REVIEW

**Review by mrahanjam — APPROVED** [review]
LGTM 🚀

File: app/cdap/components/Replicator/Create/Content/SelectTables/index.tsx [hunk 1]
```diff
@@ -706,4 +706,9 @@ class SelectTablesView extends React.PureComponent<ISelectTablesProps, ISelectTa
 
 const StyledSelectTables = withStyles(styles)(SelectTablesView);
 const SelectTables = createContextConnect(StyledSelectTables);
-export default SelectTables;
+
+// Higher Order Component wrapping class component so we can use useFeatureFlag hook
+export default ({ children, ...props }) => {
+  props.useReplicationTransformation = useFeatureFlag('replication.transformations.enabled');
+  return <SelectTables {...props} />;
```
Comment (njbriggs) [inline app/cdap/components/Replicator/Create/Content/SelectTables/index.tsx]: Nit: Should we make this
```
return <SelectTables
  {{
    ...props,
    useReplicationTransformation: useFeatureFlag('replication.transformations.enabled')
  }}
/>
```
?

I know it's short HOC but mutating `props` is still bugging me.

File: app/cdap/services/react/customHooks/useFeatureFlag.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+/**
+ * Returns true if feature if enabled, false if the flag
+ * does not exist or disabled
+ * Must wrap parent component with services/react/providers/featureFlagProvider
+ * @param name feature flag name
+ * @returns boolean
+ */
+
+export function useFeatureFlag(name: keyof IFeatureFlags | string): boolean {
```
Comment (njbriggs) [inline app/cdap/services/react/customHooks/useFeatureFlag.ts]: Maybe late to be asking this, but: Was there consideration for forcing disable of features? That is, this design represents features that are by default OFF and can be forced ON with a flag. Did we think about features that are by default ON and can be forced OFF with a flag? Or just create that behavior with a flag called `disableFoo` and an inverted test?

File: app/cdap/components/Replicator/Create/Content/SelectTables/index.tsx (hunk 1 above)
Comment (seanfeld) [reply app/cdap/components/Replicator/Create/Content/SelectTables/index.tsx]: Yeah absolutely I like that a lot better.

**Review by njbriggs — APPROVED** [review]

**Pull request merged**